theia-full-docker/Dockerfile

```diff
@@ -196,36 +231,46 @@ RUN wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add - && \
     ln -s /usr/bin/clangd-$LLVM /usr/bin/clangd
 
 # Install latest stable CMake
-ARG CMAKE_VERSION=3.18.1
+ARG CMAKE_VERSION=3.19.5
 
-RUN wget "https://github.com/Kitware/CMake/releases/download/v$CMAKE_VERSION/cmake-$CMAKE_VERSION-Linux-x86_64.sh" && \
-    chmod a+x cmake-$CMAKE_VERSION-Linux-x86_64.sh && \
-    ./cmake-$CMAKE_VERSION-Linux-x86_64.sh --prefix=/usr/ --skip-license && \
-    rm cmake-$CMAKE_VERSION-Linux-x86_64.sh
+RUN ARCH= && dpkgArch="$(dpkg --print-architecture)" \
+    && case "${dpkgArch##*-}" in \
+    amd64) ARCH='x86_64';; \
+    arm64) ARCH='aarch64';; \
+    *) echo "unsupported architecture"; exit 1 ;; \
+    esac && \
+    wget "https://github.com/Kitware/CMake/releases/download/v$CMAKE_VERSION/cmake-$CMAKE_VERSION-Linux-$ARCH.sh" && \
+    chmod a+x cmake-$CMAKE_VERSION-Linux-$ARCH.sh && \
+    ./cmake-$CMAKE_VERSION-Linux-$ARCH.sh --prefix=/usr/ --skip-license && \
+    rm cmake-$CMAKE_VERSION-Linux-$ARCH.sh
 
-# Python 2-3
+# Python 3
 RUN apt-get update \
     && apt-get install -y software-properties-common \
-    && add-apt-repository -y ppa:deadsnakes/ppa \
-    && apt-get install -y python-dev python-pip \
-    && apt-get install -y python3.8 python3-dev python3-pip \
+    && apt-get install -y python3.9 python3-dev python3-pip \
     && apt-get remove -y software-properties-common \
-    && python -m pip install --upgrade pip --user \
     && python3.8 -m pip install --upgrade pip --user \
     && pip3 install python-language-server flake8 autopep8
 
-# .NET Core SDK
-ARG DOTNET_VERSION=3.1
+# .NET Core SDK (Version of SDK Build see https://dotnet.microsoft.com/download/dotnet/3.1 )
+ARG DOTNET_VERSION=3.1.407
 # Disables .NET telemetry
 ENV DOTNET_CLI_TELEMETRY_OPTOUT 1
 # Suppresses .NET welcome message
 ENV DOTNET_SKIP_FIRST_TIME_EXPERIENCE 1
 
-RUN curl -SLO "https://packages.microsoft.com/config/ubuntu/18.04/packages-microsoft-prod.deb" \
-    && dpkg -i packages-microsoft-prod.deb \
-    && rm packages-microsoft-prod.deb \
-    && apt-get update \
-    && apt-get install -y dotnet-sdk-$DOTNET_VERSION
+RUN DOTNET_ARCH= && dpkgArch="$(dpkg --print-architecture)" \
```

Comment:
I think we might end-up missing some required packages, that would be marked as a dependency of the Debian package. 

Maybe better to keep installing with a Debian package for the main arch and have a different handling for arm64?